### PR TITLE
Std main support

### PR DESCRIPTION
**Pull request by MaxVerevkin**
Add `fn main()` support (requires specific `rust-psp` hack). Also includes `psp_std_saj` and `psp_std_net` branches.

### CODE REVIEW

File: library/std/src/rt.rs [hunk 1]
```diff
@@ -77,13 +77,17 @@ unsafe fn init(argc: isize, argv: *const *const u8) {
...
         sys::init(argc, argv);
 
-        let main_guard = sys::thread::guard::init();
-        // Next, set up the current Thread with the guard information we just
-        // created. Note that this isn't necessary in general for new threads,
-        // but we just do this to name the main thread and to give it correct
-        // info about the stack bounds.
-        let thread = Thread::new(Some(rtunwrap!(Ok, CString::new("main"))));
-        thread_info::set(main_guard, thread);
+        // FIXME: remove PSP special case
```
Comment (overdrivenpotato) [inline library/std/src/rt.rs:80]: Why is this special-cased to not run on PSP?

Comment (MaxVerevkin) [reply library/std/src/rt.rs:80]: Psp was crashing (ppsspp too). I suspect it's due to some panic. Anyway I didn't have time to investigate. Plus I don't understand what PSP's threads are. All I know is that they are not ~~concurrent~~ parallel.

File: library/std/src/sys/psp/thread.rs [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+use crate::time::Duration;
 
 pub struct Thread(Void);
 
 pub const DEFAULT_MIN_STACK_SIZE: usize = 4096;
 
 impl Thread {
     // unsafe: see thread::Builder::spawn_unchecked for safety requirements
-    pub unsafe fn new(_stack: usize, _p: Box<dyn FnOnce()>) -> io::Result<Thread> {
+    pub unsafe fn new(_stack: usize, _p: Box<dyn FnOnce()>) -> io::Result<Self> {
```
Comment (sajattack) [inline library/std/src/sys/psp/thread.rs:13]: Won't we be eventually implementing this using sceKernelStartThread?

**Comment by overdrivenpotato** [comment]
Merging for now, we'll revisit everything here before any commits are pushed to a rustc PR.

**Pull request merged**